server/sonar-server/src/test/java/org/sonar/server/computation/task/projectanalysis/component/MergeBranchComponentUuidsTest.java

```diff
@@ -0,0 +1,93 @@
+import org.sonar.db.component.ComponentDto;
+import org.sonar.db.component.ComponentTesting;
+import org.sonar.server.computation.task.projectanalysis.analysis.AnalysisMetadataHolderRule;
+import org.sonar.server.computation.task.projectanalysis.analysis.Branch;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class MergeBranchComponentUuidsTest {
+  @Rule
+  public AnalysisMetadataHolderRule analysisMetadataHolder = new AnalysisMetadataHolderRule();
+
+  @Rule
+  public DbTester db = DbTester.create();
+
+  private MergeBranchComponentUuids underTest;
+  private Branch branch = mock(Branch.class);
+
+  @Before
+  public void setUp() {
+    underTest = new MergeBranchComponentUuids(analysisMetadataHolder, db.getDbClient());
+    analysisMetadataHolder.setBranch(branch);
+  }
+
+  @Test
+  public void should_support_db_key() {
+    ComponentDto project = db.components().insertMainBranch();
+    ComponentDto mergeBranch = db.components().insertProjectBranch(project, b -> b.setKey("mergeBranch"));
+    ComponentDto branch1 = db.components().insertProjectBranch(project, b -> b.setKey("branch1"));
+    ComponentDto mergeBranchFile = ComponentTesting.newFileDto(mergeBranch, null, "file").setUuid("mergeFile");
+    ComponentDto file = ComponentTesting.newFileDto(branch1, null, "file").setUuid("file1");
+
+    db.components().insertComponents(mergeBranchFile, file);
+
+    when(branch.getMergeBranchUuid()).thenReturn(Optional.of(mergeBranch.uuid()));
+    assertThat(underTest.getUuid(file.getDbKey())).isEqualTo(mergeBranchFile.uuid());
+  }
+
+  @Test
+  public void should_support_key() {
+    ComponentDto project = db.components().insertMainBranch();
+    ComponentDto mergeBranch = db.components().insertProjectBranch(project, b -> b.setKey("mergeBranch"));
+    ComponentDto branch1 = db.components().insertProjectBranch(project, b -> b.setKey("branch1"));
+    ComponentDto mergeBranchFile = ComponentTesting.newFileDto(mergeBranch, null, "file").setUuid("mergeFile");
+    ComponentDto file = ComponentTesting.newFileDto(branch1, null, "file").setUuid("file1");
+
+    db.components().insertComponents(mergeBranchFile, file);
+
+    when(branch.getMergeBranchUuid()).thenReturn(Optional.of(mergeBranch.uuid()));
+    assertThat(underTest.getUuid(file.getKey())).isEqualTo(mergeBranchFile.uuid());
+  }
+
+  @Test
+  public void return_null_if_file_doesnt_exist() {
+    ComponentDto project = db.components().insertMainBranch();
+    ComponentDto mergeBranch = db.components().insertProjectBranch(project, b -> b.setKey("mergeBranch"));
+    ComponentDto branch1 = db.components().insertProjectBranch(project, b -> b.setKey("branch1"));
+    ComponentDto mergeBranchFile = ComponentTesting.newFileDto(mergeBranch, null, "file").setUuid("mergeFile");
+    ComponentDto file = ComponentTesting.newFileDto(branch1, null, "file").setUuid("file1");
```

Comment:
sort of, just to ensure that in this case no existing file is not returned (as in the other tests)